### PR TITLE
fix(batch-exports): Couple of backfill bugfixes

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

A couple of bugs that need squashing:
* We are not parsing heartbeat details correctly, seeing "Expected 4 but got 37", which means we are likely unpacking a full string.
* ~We are not able to cancel backfills due to the Z suffix used by temporal instead of +00:00.~

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: posthog/batch_exports/models.py [hunk 1]
```diff
@@ -292,4 +292,10 @@ def workflow_id(self) -> str:
         end_at = self.end_at and self.end_at.isoformat()
         start_at = self.start_at and self.start_at.isoformat()
 
+        if isinstance(end_at, str) and end_at.endswith("+00:00"):
+            end_at = end_at.replace("+00:00", "Z")
```
Comment (tomasfarias) [inline posthog/batch_exports/models.py:296]: By default, python seems to prefer "+00:00" but Temporal schedules everything with "Z", it's quite annoying.

File: posthog/temporal/batch_exports/backfill_batch_export.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ async def backfill_schedule(inputs: BackfillScheduleInputs) -> None:
         if details:
             # If we receive details from a previous run, it means we were restarted for some reason.
             # Let's not double-backfill and instead wait for any outstanding runs.
-            last_activity_details = HeartbeatDetails(*details[0])
+            last_activity_details = HeartbeatDetails(*details)
```
Comment (tomasfarias) [inline posthog/temporal/batch_exports/backfill_batch_export.py:176]: `details` is already the tuple recorded by temporal. By doing `details[0]` we access the first item of the tuple (`workflow_id`), which is unpacked into a bunch of strings. This is, clearly, not good.

File: posthog/batch_exports/models.py (hunk 1 above)
Comment (rossgray) [inline posthog/batch_exports/models.py:296]: looks fine, but just wondering if there's there a reason you can't just do

```python
end_at = self.end_at and self.end_at.strftime("%Y-%m-%dT%H:%M:%SZ")
```

**Review by rossgray — APPROVED** [review]
Looks good. Just wondering if you could simplify the date formatting slightly

File: posthog/batch_exports/models.py (hunk 1 above)
Comment (tomasfarias) [reply posthog/batch_exports/models.py]: I actually reverted this and copied the same formatting we use here: https://github.com/PostHog/posthog/blob/master/posthog/batch_exports/service.py#L475

Makes sense to have the service and model be consistent. But we could eventually update both.

**Review by rossgray — APPROVED** [review]

**Pull request merged**